### PR TITLE
Improve gateway records pallet

### PR DESCRIPTION
**Pull request by koushiro**
Signed-off-by: koushiro <koushiro.cqx@gmail.com>

## Types

Add `WithdrawalRecordId = u32`
Rename `WithdrawalRecordForRpc` to `RpcWithdrawalRecord`

## Storage

Rename `SerialNumber` to `NextWithdrawRecordId`

## Event

Rename `Deposit` => `Deposited`, `FinishWithdrawal` => `WithdrawalFinished`
Add `WithdrawalProcessed`, `WithdrawalRecovered`, `WithdrawalCanceled`

## Call

Rename `root_withdrawal` to `root_withdraw`

### CODE REVIEW

File: xpallets/gateway/bitcoin/src/tests/trustee.rs [hunk 1]
```diff
@@ -208,10 +208,8 @@ fn force_replace_withdraw() {
 
         let balance1 = (9778400 + withdrawal_fee).into();
         let balance2 = (9900000 + withdrawal_fee).into();
-        XGatewayRecords::deposit(
-            &alice,&X_BTC, balance1,).unwrap();
-        XGatewayRecords::deposit(
-            &bob,&X_BTC, balance2,).unwrap();
+        XGatewayRecords::deposit(&alice,X_BTC, balance1).unwrap();
+        XGatewayRecords::deposit(&bob,X_BTC, balance2).unwrap();
```
Comment (liuchengxu) [inline xpallets/gateway/bitcoin/src/tests/trustee.rs]:
```suggestion
        XGatewayRecords::deposit(&alice, X_BTC, balance1).unwrap();
        XGatewayRecords::deposit(&bob, X_BTC, balance2).unwrap();
```

File: xpallets/gateway/records/src/lib.rs [hunk 2]
```diff
@@ -3,437 +3,486 @@
...
+        <T as frame_system::Trait>::AccountId,
+        Balance = BalanceOf<T>,
+        WithdrawalRecord = WithdrawalRecordOf<T>
+    {
+        Deposit(AccountId, AssetId, Balance),
+        ApplyWithdrawal(WithdrawalRecordId, WithdrawalRecord),
+        ProcessWithdrawal(WithdrawalRecordId),
+        RecoverWithdrawal(WithdrawalRecordId),
+        CancelWithdrawal(WithdrawalRecordId, WithdrawalState),
+        FinishWithdrawal(WithdrawalRecordId, WithdrawalState),
```
Comment (liuchengxu) [inline xpallets/gateway/records/src/lib.rs]:
```suggestion
        Deposited(AccountId, AssetId, Balance),
        WithdrawalApplied(WithdrawalRecordId, WithdrawalRecord),
        WithdrawalProcessed(WithdrawalRecordId),
        WithdrawalRecovered(WithdrawalRecordId),
        WithdrawalCanceled(WithdrawalRecordId, WithdrawalState),
        WithdrawalFinished(WithdrawalRecordId, WithdrawalState),
```

File: xpallets/gateway/records/src/lib.rs [hunk 3]
```diff
@@ -3,437 +3,486 @@
...
+    }
+);
+
+decl_storage! {
+    trait Store for Module<T: Trait> as XGatewayRecords {
+        /// Withdraw applications collection, use serial numbers to mark them.
+        pub PendingWithdrawals get(fn pending_withdrawals):
+            map hasher(twox_64_concat) WithdrawalRecordId => Option<WithdrawalRecordOf<T>>;
+
+        /// The state of withdraw record corresponding to a id.
```
Comment (liuchengxu) [inline xpallets/gateway/records/src/lib.rs]:
```suggestion
        /// The state of withdraw record corresponding to an id.
```

File: xpallets/gateway/records/src/lib.rs [hunk 4]
```diff
@@ -3,437 +3,486 @@
...
+        pub WithdrawalStateOf get(fn state_of):
+            map hasher(twox_64_concat) WithdrawalRecordId => Option<WithdrawalState>;
+
+        /// The id of next withdraw record.
```
Comment (liuchengxu) [inline xpallets/gateway/records/src/lib.rs]:
```suggestion
        /// The id of next withdrawal record.
```

**Review by liuchengxu — APPROVED** [review]
Better to be also approved by @AtenJin.

**Review by atenjin — APPROVED** [review]

**Comment by koushiro** [comment]
cc @qinghuan-chain

**Pull request merged**